pages/company/visualizations/comapny_associated_activity.py

```diff
@@ -0,0 +1,256 @@
+from pages.utils.graph_utils import color_seq
+from queries.company_query import company_query as cq
+import io
+from cache_manager.cache_manager import CacheManager as cm
+from pages.utils.job_utils import nodata_graph
+import time
+import datetime as dt
+
+PAGE = "company"
+VIZ_ID = "company-associated-activity"
+
+paramter_1 = "company-contributions-required"
+paramter_2 = "checks"
+
+
+gc_compay_associated_activity = dbc.Card(
+    [
+        dbc.CardBody(
+            [
+                html.H3(
+                    "Company Associated Activity",
+                    className="card-title",
+                    style={"textAlign": "center"},
+                ),
+                dbc.Popover(
+                    [
+                        dbc.PopoverHeader("Graph Info:"),
+                        dbc.PopoverBody(
+                            "This graph counts the number of contributions that COULD be linked to each company.\n\
+                            The methodology behind this is to take each associated email to someones github account\n\
+                            and link the contributions to each as it is unknown which initity the actvity was done for."
+                        ),
+                    ],
+                    id=f"{PAGE}-popover-{VIZ_ID}",
+                    target=f"{PAGE}-popover-target-{VIZ_ID}",  # needs to be the same as dbc.Button id
+                    placement="top",
+                    is_open=False,
+                ),
+                dcc.Loading(
+                    dcc.Graph(id=VIZ_ID),
+                ),
+                dbc.Form(
+                    [
+                        dbc.Row(
+                            [
+                                dbc.Label(
+                                    "Contributions Required:",
+                                    html_for=f"{PAGE}-{paramter_1}-{VIZ_ID}",
+                                    width={"size": "auto"},
+                                ),
+                                dbc.Col(
+                                    dbc.Input(
+                                        id=f"{PAGE}-{paramter_1}-{VIZ_ID}",
+                                        type="number",
+                                        min=1,
+                                        max=100,
+                                        step=1,
+                                        value=10,
+                                        size="sm",
+                                    ),
+                                    className="me-2",
+                                    width=2,
+                                ),
+                                dbc.Col(
+                                    [
+                                        dbc.Checklist(
+                                            id=f"{PAGE}-{paramter_2}-{VIZ_ID}",
+                                            options=[
+                                                {"label": "Exclude gmail", "value": "gmail"},
```

Comment:
In a future iteration we should allow the user to ignore multiple domains that they specify- e.g. yahoo.com, outlook.com, gmail.com, etc.

Reply:
bet I can work on this now, one thing ill need help with is the layout. The time selector makes things a little weird